src/core/shaderlib/project/viewport-uniforms.js

```diff
@@ -164,7 +164,8 @@ export function getUniformsFromViewport({
   const distanceScales = viewport.getDistanceScales();
 
   // TODO - does this depend on useDevicePixels?
-  const devicePixelRatio = (window && window.devicePixelRatio) || 1;
+  const devicePixelRatio =
+    (typeof window !== 'undefined' && window && window.devicePixelRatio) || 1;
```

Comment:
`(typeof window !== 'undefined' && window &&` - you can drop the second check `window`. Either it is an object or it isn't there.